### DISCUSSION TITLE
yhat1 calculation

### DISCUSSION THREAD

**Discussion by Amjad-AbuRmileh** [ourownstory/neural_prophet#411] [Q&A - get help using NeuralProphet] [answered]
I wanted to know how is the yhat1 calculated at each timestep? does it include all the seasonal, trend, AR components aggregated?

**Comment by ourownstory** [comment, marked as answer]
Hi @Amjad-AbuRmileh Yes,  yhatX is the X-th step ahead forecast. in your case the prediction for the next value of the series.
Yes, It is the sum of all components.

**Reply by Amjad-AbuRmileh** [reply]
@ourownstory Thanks!! and the yhat1 is the closest to present, right? so its the first in the sequence of X predictions ahead?

**Reply by ourownstory** [reply]
Yes, well, in fact, the way that the forecast dataframe is returned, X describes the age of the prediction.
for a given timestamp, the prediction yhat3 refers to the predicted value for this time stamp, predicted with information available 3 steps ago.

**Reply by ourownstory** [reply]
@Amjad-AbuRmileh is the explanation introduced #413 sufficient  that you would have understood it before I answered here?
